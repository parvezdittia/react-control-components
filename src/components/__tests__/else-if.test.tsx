import React from 'react';
import renderer from 'react-test-renderer';
import { ElseIf } from '..';

describe('<ElseIf />', () => {
  it('renders the component correctly', () => {
    const component = renderer.create(
      <ElseIf condition={true}>
        <div>Child Component</div>
      </ElseIf>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the component correctly without children if condition is false', () => {
    const component = renderer.create(
      <ElseIf condition={false}>
        <div>Child Component</div>
      </ElseIf>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
