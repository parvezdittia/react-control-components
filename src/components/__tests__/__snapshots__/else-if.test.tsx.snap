// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ElseIf /> renders the component correctly 1`] = `
<div>
  Child Component
</div>
`;

exports[`<ElseIf /> renders the component correctly without children if condition is false 1`] = `null`;
